pkg/controllers/management/aks/aks_cluster_handler.go

```diff
@@ -0,0 +1,426 @@
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return cluster, err
+		}
+
+		cluster, err = e.SetUnknown(cluster, apimgmtv3.ClusterConditionWaiting, "Waiting for API to be available")
+		if err != nil {
+			return cluster, err
+		}
+
+		aksClusterConfigDynamic, err = buildAKSCCCreateObject(cluster)
+		if err != nil {
+			return cluster, err
+		}
+
+		aksClusterConfigDynamic, err = e.DynamicClient.Namespace(namespace.GlobalNamespace).Create(context.TODO(), aksClusterConfigDynamic, v1.CreateOptions{})
+		if err != nil {
+			return cluster, err
+		}
+	}
+
+	aksClusterConfigMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&cluster.Spec.AKSConfig)
+	if err != nil {
+		return cluster, err
+	}
+
+	// check for changes between aks spec on cluster and the aks spec on the aksClusterConfig object
+	if !reflect.DeepEqual(aksClusterConfigMap, aksClusterConfigDynamic.Object["spec"]) {
+		logrus.Infof("change detected for cluster [%s], updating AKSClusterConfig", cluster.Name)
+		return e.updateAKSClusterConfig(cluster, aksClusterConfigDynamic, aksClusterConfigMap)
+	}
+
+	// get aks Cluster Config's phase
+	status, _ := aksClusterConfigDynamic.Object["status"].(map[string]interface{})
+	phase, _ := status["phase"]
+	failureMessage, _ := status["failureMessage"].(string)
+
+	switch phase {
+	case "creating":
+		if cluster.Status.AKSStatus.UpstreamSpec == nil {
+			return e.setInitialUpstreamSpec(cluster)
+		}
+
+		e.ClusterEnqueueAfter(cluster.Name, enqueueTime)
+		if failureMessage == "" {
+			logrus.Infof("waiting for cluster AKS [%s] to finish creating", cluster.Name)
+			return e.SetUnknown(cluster, apimgmtv3.ClusterConditionProvisioned, "")
+		}
+		logrus.Infof("waiting for cluster AKS [%s] create failure to be resolved", cluster.Name)
+		return e.SetFalse(cluster, apimgmtv3.ClusterConditionProvisioned, failureMessage)
+	case "active":
+		if cluster.Spec.AKSConfig.Imported {
+			if cluster.Status.AKSStatus.UpstreamSpec == nil {
+				// non imported clusters will have already had upstream spec set
+				return e.setInitialUpstreamSpec(cluster)
+			}
+
+			if apimgmtv3.ClusterConditionPending.IsUnknown(cluster) {
+				cluster = cluster.DeepCopy()
+				apimgmtv3.ClusterConditionPending.True(cluster)
+				return e.ClusterClient.Update(cluster)
+			}
+		}
+
+		cluster, err = e.SetTrue(cluster, apimgmtv3.ClusterConditionProvisioned, "")
+		if err != nil {
+			return cluster, err
+		}
+
+		if cluster.Status.APIEndpoint == "" {
+			return e.RecordCAAndAPIEndpoint(cluster)
+		}
+
+		if cluster.Status.AKSStatus.PrivateRequiresTunnel == nil &&
+			to.Bool(cluster.Status.AKSStatus.UpstreamSpec.PrivateCluster) {
+			// Check to see if we can still use the control plane endpoint even though
+			// the cluster has private-only access
+			serviceToken, mustTunnel, err := e.generateSATokenWithPublicAPI(cluster)
+			if err != nil {
+				return cluster, err
+			}
+			if mustTunnel {
```

Comment:
Once the signature of `e.generateSATokenWithPublicAPI` changes, this should change as well.
```suggestion
			if mustTunnel != nil {
```